File: VB/WpfSchedulerTimeRulers/MainViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace WpfSchedulerTimeRulers

    Public Class MainViewModel
        Inherits ViewModelBase

        Private _TimeRulers As ObservableCollection(Of WpfSchedulerTimeRulers.TimeRulerViewModel)

        Public Property TimeRulers As ObservableCollection(Of TimeRulerViewModel)
            Get
                Return _TimeRulers
            End Get

            Protected Set(ByVal value As ObservableCollection(Of TimeRulerViewModel))
                _TimeRulers = value
            End Set
        End Property

        Public Sub New()
            CreateTimeRulers()
        End Sub

        Private Sub CreateTimeRulers()
            TimeRulers = New ObservableCollection(Of TimeRulerViewModel) From {New TimeRulerViewModel("GMT", TimeZoneInfo.FindSystemTimeZoneById("GMT Standard Time")), New TimeRulerViewModel("Local", TimeZoneInfo.Local)}
        End Sub
    End Class

    Public Class TimeRulerViewModel
        Inherits BindableBase

        Public Sub New(ByVal caption As String, ByVal timeZoneInfo As TimeZoneInfo)
            Me.Caption = caption
            TimeZone = timeZoneInfo
            AlwaysShowTimeDesignator = False
            ShowMinutes = False
        End Sub

        Public Property Caption As String

        Public Property TimeZone As TimeZoneInfo

        Public Property AlwaysShowTimeDesignator As Boolean

        Public Property ShowMinutes As Boolean
    End Class
End Namespace

File: CS/WpfSchedulerTimeRulers/MainViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace WpfSchedulerTimeRulers {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<TimeRulerViewModel> TimeRulers { get; protected set; }

        public MainViewModel() {
            CreateTimeRulers();
        }

        private void CreateTimeRulers() {
            TimeRulers = new ObservableCollection<TimeRulerViewModel> {
                new TimeRulerViewModel("GMT", TimeZoneInfo.FindSystemTimeZoneById("GMT Standard Time")),
                new TimeRulerViewModel("Local", TimeZoneInfo.Local)
            };
        }
    }

    public class TimeRulerViewModel : BindableBase {
        public TimeRulerViewModel(string caption, TimeZoneInfo timeZoneInfo) {
            Caption = caption;
            TimeZone = timeZoneInfo;
            AlwaysShowTimeDesignator = false;
            ShowMinutes = false;
        }
        public string Caption { get; set; }
        public TimeZoneInfo TimeZone { get; set; }
        public bool AlwaysShowTimeDesignator { get; set; }
        public bool ShowMinutes { get; set; }
    }
}
